src/gpuarray_elemwise.c

```diff
@@ -561,57 +562,55 @@ GpuElemwise *GpuElemwise_new(gpucontext *ctx,
   if (res == NULL) return NULL;
 
   res->flags = flags;
+  res->nd = 8;
+  res->n = n;
 
   res->expr = strdup(expr);
   if (res->expr == NULL)
-    goto fail_expr;
-  if (preamble == NULL) {
-    res->preamble = NULL;
-  } else {
+    goto fail;
```

Comment:
To be safe, on line 561 you should call calloc, as otherwise res isn't zeroed and if there is an error, the new free code can have problems.